### PR TITLE
Fix deoplete unicode completion

### PR DESCRIPTION
**Pull request by pappasam**
* Passes the tests

tl;dr LanguageClient-neovim's Deoplete source caused completion failures when completing strings AFTER unicode characters had been entered earlier on the same line. This PR resolves the issue.

## Symptoms

I spent the weekend writing a language server for a writing project that I used with LanguageClient-neovim and Deoplete. This language server supplies me with string values with lots of Unicode special characters, which presents an interesting edge case for this LanguageClient.

Things worked fine as long as a line began with ASCII-like characters, but broke down once special unicode characters were present:

```txt
This line is totally fine, and words complete through the end :)

This līōn will stop completing correctly after the first līōn :(
```

## Cause

I'm not 100% certain about what was happening, but my strong hunch: VimL string functions historically operated on Bytes. The get_complete_position function implementation referenced a vimscript function that uses 'match', which is one-such function. See below:

```viml
function! LanguageClient#get_complete_start(input) abort
    " echomsg a:input
    return match(a:input, '\k*$')
endfunction
```

For ASCII-like strings, bytes can be treated identically to Vim characters. When we get to the broader Unicode universe, bytes and strings behave differently in VimL.

## Fix

Delete the overriding method.

Luckily, the Deoplete plugin provides a default implementation for get_complete_position that I believe is perfectly suitable for our use-cases. No need to re-invent the wheel unless we have to!

## Additional work

I did some minor tweaks to the Deoplete plugin, cleaning up the code. I've separated the changes into their own commits; please let me know if anything is unclear.

### CODE REVIEW

**Comment by autozimu** [comment]
Have you verified this new change won't affect previous closed issues? namely,
- https://github.com/autozimu/LanguageClient-neovim/issues/427
- https://github.com/autozimu/LanguageClient-neovim/pull/456

**Comment by pappasam** [comment]
tl;dr I've manually verified that this new change won't affect the aforementioned previously closed issues.

I'm using my branch on the daily for Python / for my custom language server and have experienced no issues between the two.

#456 I tested by adding non-keyword characters to my language server; I observe no regressions to this issue. Additionally, the PR associated with issue 456 only made changes to the omnifunc implementation, LanguageClient#complete, which is separate from the deoplete fix I made.

#427 I installed the R language server and tested some examples with "." in variable names using both my branch and the current "next" branch. Neither branch experiences the issues with 427. My configuration:

```viml
\ 'r': ['R', '--slave', '-e', 'languageserver::run()'],
```

NOTE: I may have found an existing issue with R: with the R language server configured, Deoplete works fine for me, but omnicomplete (C-x C-o) works in neither the "next" branch nor the branch in this PR. Therefore, desired behavior from #427 works under deoplete but NOT with the omnifunc. Omnifunc works for for Python and my custom language server, so I imagine that 427 may be yet unfixed. A task for another day...

**Comment by autozimu** [comment]
Thank you so much for verifying!

Actually the original PR introduced `get_complete_position` override is https://github.com/autozimu/LanguageClient-neovim/pull/339

Can you check this change doesn't introduce regression regarding https://github.com/autozimu/LanguageClient-neovim/pull/339

**Comment by autozimu** [comment]
Never mind. I verified #339 is not regressed with this change.

**Comment by autozimu** [comment]
Thanks for contributing!
